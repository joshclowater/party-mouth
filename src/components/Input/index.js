import styled from 'styled-components';

export default styled.input`
  width: calc(100% - 4vh - 2px);
  padding: 1.3vh 2vh;
  font-size: 2vh;
  background-color: #fff;
  background-image: none;
  border: 1px solid;
  border-color: ${props => (props.hasError ? 'red' : '#ccc')};
  border-radius: 4px;
  box-shadow: inset 0 1px 1px rgba(0, 0, 0, 0.075);
  transition: border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
  transition-property: border-color, box-shadow;
  transition-duration: 0.15s, 0.15s;
  transition-timing-function: ease-in-out, ease-in-out;
  transition-delay: initial, initial;

  :focus {
    border-color: ${props => (props.hasError ? 'red' : '#66afe9')};
    outline: 0;
    box-shadow: inset 0 1px 1px rgba(0, 0, 0, 0.075),
      0 0 8px rgba(102, 175, 233, 0.6);
  }
`;
