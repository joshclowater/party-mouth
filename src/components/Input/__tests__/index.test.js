import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';
import Input from '../';

it('Renders correctly', () => {
  expect(shallow(<Input />)).toMatchSnapshot();
});

describe('hasError', () => {
  const wrapper = shallow(<Input hasError />);
  it('has red border', () => {
    expect(wrapper).toHaveStyleRule('border-color', 'red');
  });
  it('has red border on focus', () => {
    expect(wrapper).toHaveStyleRule('border-color', 'red', {
      modifier: ':focus'
    });
  });
});
