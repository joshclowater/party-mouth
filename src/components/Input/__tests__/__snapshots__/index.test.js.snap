// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
.c0 {
  width: calc(100% - 4vh - 2px);
  padding: 1.3vh 2vh;
  font-size: 2vh;
  background-color: #fff;
  background-image: none;
  border: 1px solid;
  border-color: #ccc;
  border-radius: 4px;
  box-shadow: inset 0 1px 1px rgba(0,0,0,0.075);
  -webkit-transition: border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out;
  transition: border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out;
  -webkit-transition-property: border-color,box-shadow;
  transition-property: border-color,box-shadow;
  -webkit-transition-duration: 0.15s,0.15s;
  transition-duration: 0.15s,0.15s;
  -webkit-transition-timing-function: ease-in-out,ease-in-out;
  transition-timing-function: ease-in-out,ease-in-out;
  -webkit-transition-delay: initial,initial;
  transition-delay: initial,initial;
}

.c0:focus {
  border-color: #66afe9;
  outline: 0;
  box-shadow: inset 0 1px 1px rgba(0,0,0,0.075), 0 0 8px rgba(102,175,233,0.6);
}

<input
  className="c0"
/>
`;
